### PR TITLE
Handle watch asset accept and reject using ApprovalController only

### PR DESCRIPTION
**Pull request by bergarces**
## Explanation

- Updates assets-controllers `TokensController`, which makes it so that only acceptance or rejection of the approval are triggered from the FE. It will no longer call `TokensController`.

Fixes https://github.com/MetaMask/MetaMask-planning/issues/463

## Manual Testing Steps

- Open Test Dapp
- Access EIP 747 section
- Try to add a new ERC20 token

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: app/scripts/lib/rpc-method-middleware/handlers/watch-asset.js
Comment (bergarces) [inline app/scripts/lib/rpc-method-middleware/handlers/watch-asset.js:46]: This is no longer needed, since that's the error we are defining in the FE component for rejection.

File: ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.js
Comment (adonesky1) [inline ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.js:16]: Presumably you'll have to repoint this selector since this state will be different as it no longer lives on the TokensController?

Comment (bergarces) [reply ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.js:16]: Good catch, we have a new selector that uses approvals filtered by type. It'll definitely need to be used here.

File: .yarn/patches/@metamask-assets-controllers-npm-6.0.0-0cb763bd07.patch
Comment (bergarces) [inline .yarn/patches/@metamask-assets-controllers-npm-6.0.0-0cb763bd07.patch:1]: Changes from this patch are already included in the new version.

File: ui/selectors/selectors.js
Comment (bergarces) [inline ui/selectors/selectors.js:507]: suggested assets already were part of pending approvals from a previous PR.

File: ui/store/actions.ts
Comment (matthewwalsh0) [inline ui/store/actions.ts:2324]: For slower connections, do we lose the loading indicator now?

If so, maybe it's worth looking at a separate ticket to add this into the `resolvePendingApproval` and `rejectPendingApproval` actions.

File: test/e2e/fixture-builder.js
Comment (matthewwalsh0) [inline test/e2e/fixture-builder.js:292]: I think I mentioned on the core PR also, but do we have a policy to remove old unused user data using migrations, or leave it to linger?

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/store/actions.ts
Comment (bergarces) [reply ui/store/actions.ts:2324]: Up until now, resolving or rejecting an approval would happen very fast.

Now that we have the `expectsResult` option though, we should probably do what you suggest and add the loading indication for resolving actions.

File: test/e2e/fixture-builder.js
Comment (bergarces) [reply test/e2e/fixture-builder.js:292]: I'm going to ask in one of the dev channels.

Comment (bergarces) [reply test/e2e/fixture-builder.js:293]: Turns out a migration to get rid of the state in existing installations would be desirable in this case.

File: app/scripts/migrations/087.ts
Comment (bergarces) [inline app/scripts/migrations/087.ts:1]: Added migration. Based on migration `085`, which has a similar use case.

**Review by vinistevam — APPROVED** [review]

**Review by mcmire — APPROVED** [review]
Patches to TokensController (at least in terms of taking care of https://github.com/MetaMask/core/pull/1261 and https://github.com/MetaMask/core/pull/1268) look good.

**Pull request merged**
